### PR TITLE
Support rejection of non-storage collections on create

### LINKED ISSUE #6: Storage Collection requires appropriate behaviour
Storage Collections currently allows behaviour for non storage collections this needs to be rejected

https://github.com/dlcs/iiif-presentation/blob/feature/updateCollection/src/IIIFPresentation/API/Features/Storage/Requests/CreateCollection.cs

Updates and Create are rejected if
"behavior" does not include "storage-collection" (see [iiif-behaviors](https://deploy-preview-3--dlcs-docs.netlify.app/api-doc/iiif#iiif-behaviors))

### PR DESCRIPTION
**Pull request by JackLewis-digirati**
This PR resolves #6 

It allows rejection of creation when `behavior` does not contain `storage-collection`

Additionally, this PR updates `CreateCollection` to use the more specific `UpsertFlatCollection` to only allow the values that can be updated

### CODE REVIEW

File: src/IIIFPresentation/API/Features/Storage/Requests/CreateCollection.cs [hunk 1]
```diff
@@ -43,7 +44,6 @@ public async Task<ModifyEntityResult<FlatCollection>> Handle(CreateCollection re
             CustomerId = request.CustomerId,
             IsPublic = request.Collection.Behavior.IsPublic(),
             IsStorageCollection = request.Collection.Behavior.IsStorageCollection(),
-            ItemsOrder = request.Collection.ItemsOrder,
```
Comment (JackLewis-digirati) [inline src/IIIFPresentation/API/Features/Storage/Requests/CreateCollection.cs:46]: not sure this is correct -m ight be worth adding this back in?

File: src/IIIFPresentation/API/Features/Storage/Validators/UpsertFlatCollectionValidator.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+namespace API.Features.Storage.Validators;
+
+public class UpsertFlatCollectionValidator : AbstractValidator<UpsertFlatCollection>
+{
+    public UpsertFlatCollectionValidator()
+    {
+        RuleFor(f => f.Parent).NotEmpty().WithMessage("Requires a 'parent' to be set");
+        RuleFor(f => f.Slug).NotEmpty().WithMessage("Requires a 'slug' to be set");
+
+        RuleFor(f => f.Behavior).Must(f => f.Contains("storage-collection"))
```
Comment (donaldgray) [inline src/IIIFPresentation/API/Features/Storage/Validators/UpsertFlatCollectionValidator.cs:13]: Should this be `Behavior.IsStorageCollection`

**Review by donaldgray — APPROVED** [review]

File: src/IIIFPresentation/API/Features/Storage/Requests/CreateCollection.cs (hunk 1 above)
Comment (donaldgray) [reply src/IIIFPresentation/API/Features/Storage/Requests/CreateCollection.cs:46]: who's best to answer that?

File: src/IIIFPresentation/Models/API/Collection/Upsert/UpsertFlatCollection.cs [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ﻿using IIIF.Presentation.V3.Strings;
 
-namespace Models.API.Collection.Update;
+namespace Models.API.Collection.Upsert;
 
-public class UpdateFlatCollection
+public class UpsertFlatCollection
 {
     public List<string> Behavior { get; set; } = new ();
```
Comment (griffri) [inline src/IIIFPresentation/Models/API/Collection/Upsert/UpsertFlatCollection.cs:7]: `new ()` should be `new()`.

**Review by griffri — APPROVED** [review]

File: src/IIIFPresentation/API/Features/Storage/Requests/CreateCollection.cs (hunk 1 above)
Comment (JackLewis-digirati) [reply src/IIIFPresentation/API/Features/Storage/Requests/CreateCollection.cs:46]: talked with Tom, decision is to add

**Pull request merged**